### PR TITLE
JWS-2579 - Refactored code to include new naming convention

### PR DESCRIPTION
**Pull request by PaulLodge**
Update to the new naming convention that will be used for natives

### CODE REVIEW

File: core/src/main/groovy/noe/ews/utils/AbstractJwsNameHelper.groovy [hunk 1]
```diff
@@ -0,0 +1,128 @@
+package noe.ews.utils
+
+import groovy.util.logging.Slf4j
+import noe.common.utils.Platform
+import noe.common.utils.Version
+
+/**
+ * JWS Name helper provides logic behind JWS ZIP filenames. These filenames differ based on the EWS version, as well as
+ * old vs new naming. If you require old naming, toggle compatibility mode. Otherwise, new naming will be provided.
+ * Old naming example: jws-application-server-5.0.0-RHEL7-x86_64.zip
```
Comment (AdamKrajcik) [inline core/src/main/groovy/noe/ews/utils/AbstractJwsNameHelper.groovy:10]: Maybe add the new natives naming here as well?

**Review by AdamKrajcik — CHANGES_REQUESTED** [review]
I've added some comments. Nothing really bad stands out to me.

File: core/src/main/groovy/noe/ews/utils/JwsNameHelper.groovy [hunk 2]
```diff
@@ -12,119 +11,59 @@ import noe.common.utils.Version
...
-    archSeparator = "."
+    if (version.getMajorVersion() <= 5) {
+      helper = new JwsNameHelper_Legacy(version)
+    } else {
+      helper = new JwsNameHelper_Current(version)
+    }
+  }
+
+  void setArchSeparator(String archSeparator) {
+    helper.archSeparator = archSeparator
```
[the remark this answers is not in the document]
Comment (PaulLodge) [reply core/src/main/groovy/noe/ews/utils/JwsNameHelper.groovy:25]: I think was simply an oversight on my part

File: core/src/main/groovy/noe/ews/utils/AbstractJwsNameHelper.groovy (hunk 1 above)
Comment (PaulLodge) [reply core/src/main/groovy/noe/ews/utils/AbstractJwsNameHelper.groovy:10]: I think here it can be skipped

**Review by AdamKrajcik — COMMENTED** [review]
I added a few non-blocking comments.

File: core/src/main/groovy/noe/ews/utils/JwsNameHelper_Current.groovy [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ */
+@Slf4j
+class JwsNameHelper_Current extends AbstractJwsNameHelper{
+
+  JwsNameHelper_Current(Version version) {
+    this.version = version
+    this.jwsMajorVersion = version.majorVersion
+    this.platform = new Platform()
+    productName = "jws"
+    archSeparator = "."
```
[the remark this answers is not in the document]
Comment (PaulLodge) [reply core/src/main/groovy/noe/ews/utils/JwsNameHelper_Current.groovy]: oops yep you're right

File: core/src/main/groovy/noe/ews/utils/JwsNameHelper_Legacy.groovy [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ */
+@Slf4j
+class JwsNameHelper_Legacy extends AbstractJwsNameHelper{
+
+  JwsNameHelper_Legacy(Version version) {
+    this.version = version
+    this.jwsMajorVersion = version.majorVersion
+    this.platform = new Platform()
+    productName = jwsMajorVersion <= 2 ? "jboss-ews" : "jws"
+    archSeparator = "."
```
[the remark this answers is not in the document]
Comment (PaulLodge) [reply core/src/main/groovy/noe/ews/utils/JwsNameHelper_Legacy.groovy]: oops yep you're right

**Review by AdamKrajcik — APPROVED** [review]
LGTM.

**Pull request merged**